Source/Core/AudioCommon/OpenALStream.cpp

```diff
@@ -149,10 +155,13 @@ void OpenALStream::SoundLoop()
   memset(uiBuffers, 0, numBuffers * sizeof(ALuint));
   uiSource = 0;
 
-  // Checks if a X-Fi is being used. If it is, disable FLOAT32 support as this sound card has no
-  // support for it even though it reports it does.
+  if (alIsExtensionPresent("AL_EXT_float32"))
+    float32_capable = true;
+
+  // Checks if a X-Fi is being used. If it is, enable FIXED32 support as this sound card
+  // is the only one that supports it for now.
   if (strstr(alGetString(AL_RENDERER), "X-Fi"))
-    float32_capable = false;
+    fixed32_capable = true;
```

Comment:
I believe the nForce implementation is quite dead since... 2004? I bet it doesn't even work on Vista or newer. Forget about any 64-bit processor support.

ASUS seems to use the implementation from C-Media, I hope someone can test it but I bet that it only supports 16-bit fixed point, with luck it also supports floating point. It is probably software rendered and properly fall back to 16-bit if floating point support is not found. It would be great if someone could test it. Also, does it have 64-bit processor support?

My onboard audio is from Realtek, and no OpenAL driver comes with the newest driver. Is it still used/supported by Realtek?

Some Creative cards use software rendered OpenAL from Sensaura, through Sens_oal.dll driver. Last time I tested it didn't support any kind of 32-bit samples, but works fine with the 16-bit fall back.

I believe that valid software rendered drivers that should be tested are "OpenAL Soft", "AeonWave-OpenAL", "Rapture3D" and the "Generic software" from default wrap_oal.dll driver.
OpenAL Soft works great with floating point samples as anyone can test it and "Generic software" from wrap_oal.dll properly falls back to 16-bit, but it would be great if someone could test with those other renderers.